src/typescript/frontend/src/app/dexscreener/events.ts

```diff
@@ -0,0 +1,305 @@
+ //       },
+ //       "eventType": "swap",
+ //       "txnId": "0xe9e91f1ee4b56c0df2e9f06c2b8c27c6076195a88a7b8537ba8313d80e6f124e",
+ //       "txnIndex": 4,
+ //       "eventIndex": 3,
+ //       "maker": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
+ //       "pairId": "0x123",
+ //       "asset0In": 10000,
+ //       "asset1Out": 20000,
+ //       "priceNative": 2,
+ //       "reserves": {
+ //         "asset0": 500,
+ //         "asset1": 1000
+ //       }
+ //     },
+ //     {
+ //       "block": {
+ //         "blockNumber": 10,
+ //         "blockTimestamp": 1673319600
+ //       },
+ //       "eventType": "join",
+ //       "txnId": "0xea1093d492a1dcb1bef708f771a99a96ff05dcab81ca76c31940300177fcf49f",
+ //       "txnIndex": 0,
+ //       "eventIndex": 0,
+ //       "maker": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
+ //       "pairId": "0x456",
+ //       "amount0": 10,
+ //       "amount1": 5,
+ //       "reserves": {
+ //         "asset0": 100,
+ //         "asset1": 50
+ //       }
+ //     },
+ //     {
+ //       "block": {
+ //         "blockNumber": 11,
+ //         "blockTimestamp": 1673406000
+ //       },
+ //       "eventType": "swap",
+ //       "txnId": "0xea1093d492a1dcb1bef708f771a99a96ff05dcab81ca76c31940300177fcf49f",
+ //       "txnIndex": 1,
+ //       "eventIndex": 20,
+ //       "maker": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
+ //       "pairId": "0x456",
+ //       "asset0In": 0.0123456789,
+ //       "asset1Out": 0.000123456789,
+ //       "priceNative": 0.00000012345,
+ //       "reserves": {
+ //         "asset0": 0.0001,
+ //         "asset1": 0.000000000000001
+ //       }
+ //     }
+ //   ]
+ // }
+ **/
+
+import { type NextRequest, NextResponse } from "next/server";
+import { type Block } from "./latest-block";
+import { fetchLiquidityEventsByBlock, fetchSwapEventsByBlock } from "@sdk/indexer-v2/queries/app/dexscreener";
+import { type toLiquidityEventModel, type toSwapEventModel } from "@sdk/indexer-v2/types";
+import Big from "big.js";
+
+
+/**
+ * - `txnId` is a transaction identifier such as a transaction hash
+ * - `txnIndex` refers to the order of a transaction within a block, the higher the index the later in the block the
+ * transaction was processed
+ * - `eventIndex` refer to the order of the event within a transaction, the higher the index the later in the
+ * transaction the event was processed
+ * - The combination of `txnIndex` + `eventIndex` must be unique to any given event within a block, including for
+ * different event types
+ * - `maker` is an identifier for the account responsible for submitting the transaction. In most cases the transaction
+ * is submitted by the same account that sent/received tokens, but in cases where this doesn't apply (i.e.: transaction
+ * submitted by an aggregator) an effort should be made to identify the underlying account
+ * - All amounts (`assetIn`/`assetOut`/`reserve`) should be decimalized (`amount / (10 ** assetDecimals)`)
```

Comment:
Just saw this, this means we need to wrap all numeric coin output values (quoteVolume, baseVolume, netProceeds, reserve.base, reserves.quote, etc) in here with `toCoinDecimalString(..., numDecimalsWeWantToUse);`

Reply:
Not sure if we need to return an actual number `0.00000123` or `"0.00000123"`, though

Reply:
For the sake of weird JS float handling, a string- this is a very good catch!